test/integration/resourcemanager/managedresource/reconciler_test.go

```diff
@@ -254,6 +309,513 @@ var _ = Describe("ManagedResource controller tests", func() {
+				}, time.Minute, time.Second).Should(BeTrue())
+
+				Consistently(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(BeNotFoundError())
+				}, 10*time.Second, time.Second).Should(Succeed())
+			})
+		})
+
+		Context("#Keep Object", func() {
+			BeforeEach(func() {
+				configMap.SetAnnotations(map[string]string{resourcesv1alpha1.KeepObject: "true"})
+
+				data, err := createSecretDataFromObject(configMap, "config-map.yaml")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(data).ToNot(BeNil())
+
+				secretForManagedResource.Data = data
+			})
+
+			JustBeforeEach(func() {
+				Expect(testClient.Create(ctx, secretForManagedResource)).To(Succeed())
+				Expect(testClient.Create(ctx, managedResource)).To(Succeed())
+
+				Eventually(func(g Gomega) bool {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return condition != nil && condition.Status == gardencorev1beta1.ConditionTrue
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+
+			It("should keep the object in case it is removed from the MangedResource", func() {
+				managedResource.Spec.SecretRefs = []corev1.LocalObjectReference{}
+				metav1.SetMetaDataAnnotation(&managedResource.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
+
+				Expect(testClient.Update(ctx, managedResource)).To(Succeed())
+
+				Eventually(func(g Gomega) bool {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+					condition := gardenerv1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return condition != nil && condition.Status == gardencorev1beta1.ConditionTrue
+				}, time.Minute, time.Second).Should(BeTrue())
+
+				Consistently(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(Succeed())
+				}, 10*time.Second, time.Second).Should(Succeed())
+			})
+
+			It("should keep the object even after deletion of ManagedResource", func() {
+				Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+				Eventually(func(g Gomega) {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+				}, time.Minute, 5*time.Second).Should(Succeed())
+
+				Expect(testClient.Delete(ctx, secretForManagedResource)).To(Or(Succeed(), BeNotFoundError()))
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(Succeed())
+			})
+		})
+
+		Context("#Ignore", func() {
+			BeforeEach(func() {
+				configMap.SetAnnotations(map[string]string{resourcesv1alpha1.Ignore: "true"})
```

Comment:
Why is the resource annotated?
Shouldn't we annotate the ManagedResource itself?

Reply:
No.

Reply:
Check this https://github.com/gardener/gardener/pull/5556

Reply:
Ref: https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#preventing-reconciliations

Reply:
Got it

Reply:
This is also right and expected test https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#ignoring-updates